### DISCUSSION TITLE
Add the ability to do custom range mapping via Bezier spline

### DISCUSSION THREAD

**Discussion by franckaubry** [cables-gl/cables_docs#123] [Feature Requests]
Add the ability to do custom range mapping via Bezier spline, resulting in a very visual way to manage data.
This is a small and normal UI feature in other software, but it gives incredible control to the range mapping of data. (both single and for arrays)
Sometimes (or very often, in my case) you don't want the range to go from min to max only, but via a completely custom range.
 
Here are some UI examples from Cinema 4D and Houdini. The two last examples gives a good idea of  why this can be useful:
![image](https://user-images.githubusercontent.com/97125854/149638928-1590bb79-bcf7-494f-828c-1ee5ad4ef7b5.png)
![image](https://user-images.githubusercontent.com/97125854/149639098-e59c6977-dfe2-488b-82c3-4fe283a355eb.png)
![image](https://user-images.githubusercontent.com/97125854/149638968-f6efc9a8-796e-487e-bfc9-cd2e88953913.png)

If one wants to make it easy for the user, a number of default spline shapes can be selected via a dropdown menu (linear, smooth, sine, ...) resulting in generating the correct curve in the spline UI.

**Comment by franckaubry** [comment]
The idea would be to change the current “Easing” drop down, into a more flexible Bézier curve.
From this:
![image](https://user-images.githubusercontent.com/97125854/149672630-40bd8370-3259-48a5-aa66-722eab9cfeea.png)
To something like this:
![image](https://user-images.githubusercontent.com/97125854/149672636-2975cba9-b734-4f32-998c-86218765b014.png)

**Comment by franckaubry** [comment]
In the Discord, this was one of the possible adaptation models:
[https://jordi.platinum.edu.pl/xsplines/](url)

![image](https://user-images.githubusercontent.com/97125854/149672769-162d4710-f8c9-413f-8ecd-ec1ba3ca5431.png)

**Comment by pandrr** [comment]
we wanted this for a long time, but no time to implement yet.
so no priority for now

**Reply by franckaubry** [reply]
Ok. Thanks for the reply. Very understandable!
I am very much looking forward to the day this will be implemented. Maybe simultaneously as the new timeline is some functions are the same?
